Implemented Sign in and Sign up

Jesil-OT opened the pull request.

Jesil-OT committed:
Implemented Sign in and Sign up

segunfrancis commented on app/src/main/java/com/sdn/teampredators/polima/di/LocalModule.kt:
```diff
@@ -2,12 +2,17 @@ package com.sdn.teampredators.polima.di
 
 import android.content.Context
 import android.content.SharedPreferences
+import com.google.firebase.auth.FirebaseAuth
```
Delete all these unused imports

segunfrancis commented on app/src/main/java/com/sdn/teampredators/polima/utils/ViewExt.kt:
```diff
@@ -21,6 +22,10 @@ private fun requestManger() =
     this.isVisible = state
+}
+
+fun View.message(msg: String){
```
This is a method not a variable. Change the name to something that depicts a method like `displayMessage` or `showMessage` or something more descriptive

segunfrancis commented on app/src/main/res/layout/fragment_sign_in.xml:
```diff
@@ -0,0 +1,165 @@
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/ti_password"
+            style="@style/itStyle"
```
Give a better name for the style. `itStyle` isn't descriptive. Follow the existing convention

segunfrancis commented on app/src/main/res/layout/fragment_sign_in.xml:
```diff
@@ -0,0 +1,165 @@
+
+            <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/et_password"
+                style="@style/etStyle"
```
Give a better name for the style. `etStyle` isn't descriptive. Follow the existing convention

segunfrancis commented on app/src/main/res/layout/fragment_sign_in.xml:
```diff
@@ -0,0 +1,165 @@
+            style="@style/TextAppearance.AppCompat.Body1"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="24dp"
```
use values from `dimens.xml`. Create them if they don't exist

segunfrancis commented on app/src/main/res/layout/fragment_sign_in.xml:
```diff
@@ -0,0 +1,165 @@
+            android:fontFamily="@font/muli"
+            android:text="@string/no_account"
+            android:textColor="@color/black"
+            android:textSize="17sp"
```
Avoid using odd numbers for dimensions and sizes, that's according to material design standards

segunfrancis commented on app/src/main/res/layout/fragment_sign_in.xml:
```diff
@@ -0,0 +1,165 @@
+            app:layout_constraintEnd_toEndOf="@+id/sign_in_button"
+            app:layout_constraintStart_toStartOf="@+id/sign_in_button" />
+
+        <com.google.android.material.textview.MaterialTextView
```
I'm sure this text family must have been used in several parts of the app, extract the styles so it can be easily reused in other parts of the app

segunfrancis commented on app/src/main/res/layout/fragment_sign_up.xml:
```diff
@@ -1,18 +1,186 @@
+        android:layout_marginBottom="24dp"
+        android:enabled="false"
+        android:text="@string/sign_up"
+        app:elevation="0dp"
```
I don't think this elevation is necessary

segunfrancis commented on app/src/main/res/layout/layout_progress_bar.xml:
```diff
@@ -0,0 +1,18 @@
+        android:layout_gravity="center"
+        android:indeterminate="true"
+        android:mirrorForRtl="false"
+        app:indicatorSize="30dp"
```
extract to `dimens.xml` file

segunfrancis commented on app/src/main/res/values/themes.xml:
```diff
@@ -54,4 +54,30 @@
+        <item name="android:textSize">14sp</item>
+    </style>
+
+    <style name="itStyle" parent="Widget.MaterialComponents.TextInputLayout.OutlinedBox.Dense">
```
Follow the naming convention, please

segunfrancis commented on app/src/main/res/values/themes.xml:
```diff
@@ -54,4 +54,30 @@
+        <item name="helperTextTextColor">@color/black</item>
+    </style>
+
+    <style name="etStyle">
```
Follow the naming convention, please

Jesil-OT committed:
- Deleted unused import statement
- Changed `message` to `Show Message` in `ViewUtils.kt`
- Changed `itStyle` to `TextInputBox` in `Styles.xml`
- Extracted all dimensions to `dimens.xml`
- Removed unnecessary elevations in `fragment_sign_in` and `fragment_sign_up` button
